packages/flutter/lib/src/widgets/implicit_animations.dart

```diff
@@ -565,6 +578,258 @@ abstract class AnimatedWidgetBaseState<T extends ImplicitlyAnimatedWidget> exten
+///     super.child,
+///   }) : super(size, lerp: Size.lerp);
+///
+///   @override
+///   Widget build(BuildContext context, Size value) {
+///     return SizedBox.fromSize(size: value, child: child);
+///   }
+/// }
+/// ```
+/// {@end-tool}
+///
+/// {@template flutter.widgets.AnimatedValue.initialValue}
+/// When the widget is initialized, it immediately animates from the
+/// [initialValue] to the [value], if an initial value is specified.
+/// Otherwise, the widget does not animate at first, but will transition
+/// each time its [value] changes.
+/// {@endtemplate}
+///
+/// {@tool dartpad}
+/// This example shows how to run and restart an animation without needing to
+/// set up an [AnimationController] or interact with it directly.
+///
+/// ** See code in examples/api/lib/widgets/implicit_animations/animated_value.0.dart **
+/// {@end-tool}
+///
+/// Changing the [duration] or the [curve] while an animation is running
+/// may cause abrupt visual changes. There are 2 ways to prevent this problem:
+///
+///  1. Wait until an animation completes before adjusting its duration or curve.
+///  2. If the [duration] and/or [curve] need to change immediately,
+///     consider changing the [value] as well. This will start an animation to
+///     the new value, using the current value as a starting point.
+///
+/// See also:
+///
+///  * [ImplicitlyAnimatedWidget], which was used to create implicit animations
+///    before [Record] types were supported in Dart.
+///  * [TweenAnimationBuilder], which is similar to [AnimatedValue.builder],
+///    but uses a [Tween] rather than a [LerpCallback] for evaluation.
+///  * [AnimatedOpacity], which is an implicitly animated version of [Opacity].
+///  * [AnimatedCrossFade], which cross-fades between two given children and
+///    animates itself between their sizes.
+///  * [AnimatedSize], which automatically transitions its size over a given
+///    duration.
+///  * [AnimatedSwitcher], which fades from one widget to another.
+abstract class AnimatedValue<T extends Object> extends StatefulWidget {
+  /// Used by subclasses to initialize fields.
+  ///
+  /// For an animated value that doesn't need an additional class declaration,
+  /// consider [AnimatedValue.builder].
+  const AnimatedValue(
+    this.value, {
+    super.key,
+    this.initialValue,
+    required this.duration,
+    this.curve = Curves.linear,
+    required this.lerp,
+    this.child,
+    this.onEnd,
+  });
+
+  /// Creates an implicit animation using the provided [Duration],
+  /// [ValueWidgetBuilder], and [LerpCallback].
+  ///
+  /// {@macro flutter.widgets.AnimatedValue.initialValue}
+  ///
+  /// See also:
+  ///
+  ///  * [TweenAnimationBuilder], which does the same,
+  ///    using a [Tween] rather than a [LerpCallback].
+  ///  * [Builder], the [StatelessWidget] equivalent.
+  const factory AnimatedValue.builder(
+    T value, {
+    Key? key,
+    T? initialValue,
+    required Duration duration,
+    Curve curve,
```

Comment:
It seems currently the analyser/editor isn't smart enough to show default value for redirected factory constructor, perhaps you can note the default value of curve in `AnimatedValue.builder` document as well?

Reply:
SGTM, thanks for catching this!